core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -0,0 +1,191 @@
+    case Assign(tps) =>
+      val previousAssigned = consumer.assignment()
+      consumer.assign((tps.toSeq ++ previousAssigned.asScala).asJava)
+    case AssignWithOffset(tps) =>
+      consumer.assign(tps.keys.toSeq.asJava)
+      tps.foreach {
+        case (tp, offset) => consumer.seek(tp, offset)
+      }
+    case Commit(offsets) =>
+      val commitMap = offsets.mapValues(new OffsetAndMetadata(_))
+      val reply = sender()
+      commitsInProgress += 1
+      consumer.commitAsync(commitMap.asJava, new OffsetCommitCallback {
+        override def onComplete(offsets: util.Map[TopicPartition, OffsetAndMetadata], exception: Exception): Unit = {
+          commitsInProgress -= 1
+          if (exception != null) reply ! Status.Failure(exception)
+          else reply ! Committed(offsets.asScala.toMap)
+        }
+      })
+      //right now we can not store commits in consumer - https://issues.apache.org/jira/browse/KAFKA-3412
+      pollExpected = true
+      poll()
+    case Subscribe(topics, listener) =>
+      consumer.subscribe(topics.toList.asJava, listener)
+    case SubscribePattern(pattern, listener) =>
+      consumer.subscribe(Pattern.compile(pattern), listener)
+    case Poll =>
+      pollExpected = true
+      poll()
+    case RequestMessages(topics) =>
+      requests ++= topics.map(_ -> sender()).toMap
+      pollExpected = true
+      poll()
+    case Stop =>
+      if (commitsInProgress == 0) {
+        context.stop(self)
+      }
+      else {
+        stopInProgress = true
+        context.become(stopping)
+      }
+  }
+
+  def stopping: Receive = LoggingReceive {
+    case Watch => sender() ! finished.future
+    case Assign(tps) =>
+    case AssignWithOffset(tps) =>
+    case Commit(offsets) =>
+      sender() ! Status.Failure(StoppingException())
+    case Subscribe(topics, listener) =>
+    case SubscribePattern(pattern, listener) =>
+    case Poll =>
+      pollExpected = true
+      poll()
+    case RequestMessages(topics) =>
+      sender() ! Status.Failure(StoppingException())
+    case Stop =>
+  }
+
+  override def preStart(): Unit = {
+    super.preStart()
+    requests = Map.empty[TopicPartition, ActorRef]
+    consumer = settings.createKafkaConsumer()
+    nextScheduledPoll = None
+    commitsInProgress = 0
+    pollExpected = false
+    finished = Promise()
+    schedulePoll()
+  }
+
+  override def postStop(): Unit = {
+    nextScheduledPoll.foreach(_.cancel())
+    consumer.close()
+    finished.trySuccess(Done)
+    super.postStop()
+  }
+
+  def poll() = {
+    if (pollExpected) {
+      pollExpected = false
+      nextScheduledPoll.foreach(_.cancel())
+      nextScheduledPoll = None
+
+      //set partitions to fetch
+      val partitionsToFetch = requests.keys.toSet
+      consumer.assignment().asScala.foreach { tp =>
+        if (partitionsToFetch.contains(tp)) consumer.resume(tp)
+        else consumer.pause(tp)
+      }
+
+      val rawResult = consumer.poll(pollTimeout().toMillis)
```

Comment:
I will not work, because `subscribe` will be applied only on a next poll. So, `subscribe` means nothing without `poll`. That is how kafka client works.